packages/components/src/components/cardradio/cardradio.component.ts

```diff
@@ -0,0 +1,134 @@
+import { CSSResult, html, nothing, PropertyValues } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './cardradio.styles';
+import { DisabledMixin } from '../../utils/mixins/DisabledMixin';
+import { TabIndexMixin } from '../../utils/mixins/TabIndexMixin';
+import Card from '../card/card.component';
+
+/**
+ * cardradio component extends `mdc-card` and supports radio selection interaction addtionally.
+ *
+ * While using this component within a form or group of cards, make sure cards are in a role = "radio-group".
+ * This card would have events for selected and unselected (similar to radio)
+ *
+ * **Note**: This is a single selection card i.e. interacting anywhere on the card would toggle the checked state.
+ * Make sure to pass only non-interactable elements within the slots.
+ *
+ * Make sure to pass the `card-title` mandatorily for this card.
+ *
+ * @tagname mdc-cardradio
+ *
+ * @dependency mdc-icon
+ * @dependency mdc-staticradio
+ * @dependency mdc-text
+ *
+ * @slot before-body - This slot is for passing the content before the body
+ * @slot body - This slot is for passing the text content for the card
+ * @slot after-body - This slot is for passing the content after the body
+ *
+ * @event click - (React: onClick) Event that gets dispatched when the card is clicked. It toggles the checked state.
+ * @event keydown - (React: onKeyDown) This event is dispatched when a key is pressed down on the card.
+ * It toggles the checked state when enter key is used.
+ * @event keyup - (React: onKeyUp) This event is dispatched when a key is released on the card.
+ * It toggles the checked state when space key is used.
+ * @event change - (React: onChange) Event that gets dispatched when the card state changes.
+ * @event focus - (React: onFocus) Event that gets dispatched when the card receives focus.
+ *
+ */
+class CardRadio extends DisabledMixin(TabIndexMixin(Card)) {
+ /**
+   * The checked state of the card
+   * @default false
+   */
+ @property({ type: Boolean, reflect: true })
+ checked: boolean = false;
```

Comment:
For primitive types, we don't need to define the type when you have a default value assigned.
```suggestion
 checked = false;
```